scripts/openapi-convert/object-fixes.mjs

```diff
@@ -0,0 +1,47 @@
+"use strict";
+import { URL } from 'node:url';
+
+// Refactor `servers` field to be able to access `basePath` easily.
+function refactorServers(obj) {
+    if (!obj.servers) {
+        return;
+    }
+
+    let server = {
+        url: "",
+        variables: {
+            basePath: {
+                default: ""
+            }
+        }
+    }
+
+    const url = new URL(obj.servers[0].url);
+
+    if (obj.servers.length > 1) {
+        // In our case several URLs always mean both http and https for the same
+        // host.
+        server.variables.protocol = {
+            enum: ["http", "https"],
+            default: "https",
+        }
+        obj.servers.pop();
+
+        server.url = "{protocol}://" + url.host + "{basePath}";
+    } else {
+        server.url = url.origin + "{basePath}";
+    }
```

Comment:
Answering to myself - insecure HTTP would not be wanted in server-server federation. Still, it seems to me that we should rather impose one specific set of schemes for each API, instead of hunting after deviations in individual files.

Reply:
Yes, I think there may have been some idea that we would never want to run the identity service, or the federation protocol, over http. Still, it's not obvious to me that the api files are a good place to try to enforce that.

Reply:
I think that the whole authority+base path piece should be out of the API definition scope: the thing is supposed to be generic. I wonder if there's a mechanism to have that part of the URL as some sort of a variable that is substituted per installation.

Reply:
We could definitely do like the published API definitions, where the URL is defined as `https://{homeserver_url}` with a default value of `matrix-client.matrix.org` for the variable (I think it's mostly a convenience to have a default server to try requests in the API viewer). The scheme could also be left out as a choice entirely.

It seems the viewer at https://matrix.org/docs/api/ enforces the server to be ` https://matrix-client.matrix.org/`, I don't know if it's on purpose. If I try the RapiDoc viewer [here](https://rapidocweb.com/examples/example1.html) with the [latest client-server API definition](https://spec.matrix.org/latest/client-server-api/api.json) it offers me to change the default URL.

Reply:
Ok, I'll make a separate suggestion in a minute.